internal/provider/resource_group_members.go

```diff
@@ -0,0 +1,358 @@
+package googleworkspace
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	directory "google.golang.org/api/admin/directory/v1"
+)
+
+type MemberChange struct {
+	Old, New map[string]interface{}
+}
+
+func resourceGroupMembers() *schema.Resource {
+	return &schema.Resource{
+		// This description is used by the documentation generator and the language server.
+		Description: "Group resource manages Google Workspace Groups.",
+
+		CreateContext: resourceGroupMembersCreate,
+		ReadContext:   resourceGroupMembersRead,
+		UpdateContext: resourceGroupMembersUpdate,
+		DeleteContext: resourceGroupMembersDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceGroupMembersImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"group_id": {
+				Description: "Identifies the group in the API request. The value can be the group's email address, " +
+					"group alias, or the unique group ID.",
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"members": {
```

Comment:
I think we'll want to set the hash here, you can copy over this [file](https://github.com/hashicorp/terraform-provider-google/blob/master/google/hashcode.go) from the google provider then add something like [this](https://github.com/hashicorp/terraform-provider-google/blob/master/google/resource_dns_managed_zone.go#L159), I used `raw["email"]` when I was playing around with it.

Reply:
I apologize for not saying anything yesterday. I'm wondering if we would be better off making this a `TypeList` and writing a `diffSuppressFunc` to suppress any diffs on order? `TypeSet` can sometimes be a nuisance to maintain, and in this case, the diffSuppress might be easier. Do you have thoughts for/against this idea?

Reply:
Will this work with `TypeList`? https://github.com/hashicorp/terraform-plugin-sdk/issues/477
The function is giving me strings and not entries from a list. How will this work? Do you have an example?

I used `TypeSet` to fix the issue with the unordered list